x-pack/plugins/observability_solution/entity_manager/server/lib/entities/built_in/services.ts

```diff
@@ -8,61 +8,112 @@
-  },
-  identityFields: ['service.name', { field: 'service.environment', optional: true }],
-  displayNameTemplate: '{{service.name}}{{#service.environment}}:{{.}}{{/service.environment}}',
-  metadata: [
-    'data_stream.type',
-    'service.instance.id',
-    'service.namespace',
-    'service.version',
-    'service.runtime.name',
-    'service.runtime.version',
-    'service.node.name',
-    'service.language.name',
-    'agent.name',
-    'cloud.provider',
-    'cloud.instance.id',
-    'cloud.availability_zone',
-    'cloud.instance.name',
-    'cloud.machine.type',
-    'host.name',
-    'container.id',
-  ],
-  metrics: [
-    {
-      name: 'logRate',
-      equation: 'A / 5',
-      metrics: [
-        {
-          name: 'A',
-          aggregation: 'doc_count',
-          filter: 'log.level: *',
-        },
-      ],
+export const builtInServicesFromLogsEntityDefinition: EntityDefinition =
+  entityDefinitionSchema.parse({
+    id: `${BUILT_IN_ID_PREFIX}services_from_ecs_data`,
+    name: 'Services from ECS data',
+    description:
+      'This definition extracts service entities from common data streams by looking for the ECS field service.name',
+    type: 'service',
+    managed: true,
+    filter: '@timestamp >= now-10m',
+    indexPatterns: ['logs-*', 'filebeat*', 'apm*', 'metrics-*'],
+    history: {
+      timestampField: '@timestamp',
+      interval: '1m',
+      settings: {
+        frequency: '2m',
+        syncDelay: '2m',
+      },
     },
-    {
-      name: 'errorRate',
-      equation: 'A / 5',
-      metrics: [
-        {
-          name: 'A',
-          aggregation: 'doc_count',
-          filter: 'log.level: "ERROR"',
-        },
-      ],
+    latest: {
+      lookback: '5m',
     },
-  ],
-});
+    identityFields: ['service.name', { field: 'service.environment', optional: true }],
+    displayNameTemplate: '{{service.name}}{{#service.environment}}:{{.}}{{/service.environment}}',
+    metadata: [
+      'data_stream.type',
+      'service.environment',
+      'service.name',
+      'service.namespace',
+      'service.version',
+      'service.runtime.name',
+      'service.runtime.version',
+      'service.language.name',
+      'cloud.provider',
+      'cloud.availability_zone',
+      'cloud.machine.type',
+    ],
+    metrics: [
+      {
+        name: 'latency',
+        equation: 'A',
+        metrics: [
+          {
+            name: 'A',
+            aggregation: 'avg',
+            filter:
+              'processor.event: "metric" AND metricset.name: "transaction" AND data_stream.dataset: "apm.transaction.1m"',
+            field: 'transaction.duration.histogram',
+          },
+        ],
+      },
+      {
+        name: 'throughput',
+        equation: 'A',
+        metrics: [
+          {
+            name: 'A',
+            aggregation: 'doc_count',
+            filter:
+              'processor.event: "metric" AND metricset.name: "transaction" AND data_stream.dataset: "apm.transaction.1m"',
```

Comment:
If you don't care about the `transaction.name` dimension you should use [service transaction metrics](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/dev_docs/apm_queries.md#transactions-in-service-inventory-page)

```suggestion
              'processor.event: "metric" AND metricset.name: "service_transaction" AND metricset.interval: "1m"',
```

Reply:
There's also service summary documents. They don't contain latency metrics but will contain `agent.name`, `
`service.environment` and obviously `service.name`.